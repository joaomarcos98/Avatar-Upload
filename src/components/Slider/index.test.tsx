/** @name Dependencies */
import { fireEvent, screen } from "@testing-library/react"
import userEvent from "@testing-library/user-event"
/** @name Components */
import { Slider } from "."
/** @name Utils */
import { renderWithTheme } from "@utils/test/renderWithTheme"

const onInputStub = jest.fn()

describe("Slider component", () => {
    it("should be renders correctly", () => {
        const { container } = renderWithTheme(<Slider />)

        expect(screen.getByRole("slider")).toBeInTheDocument()
        expect(container.firstChild).toMatchSnapshot()
    })

    it("should be render with default value", () => {
        renderWithTheme(<Slider />)

        expect(screen.getByRole("slider")).toHaveValue("20")
    })

    it("should change value", () => {
        renderWithTheme(<Slider onInput={onInputStub} />)

        const slider = screen.getByRole("slider")

        expect(slider).toHaveValue("20")
        fireEvent.change(slider, { target: { value: "30" } })
        expect(slider).toHaveValue("30")
        expect(onInputStub).toHaveBeenCalledWith("30")
    })

    it("should be accessible with tab", () => {
        renderWithTheme(<Slider />)

        expect(document.body).toHaveFocus()
        userEvent.tab()
        expect(screen.getByRole("slider")).toHaveFocus()
    })
})
