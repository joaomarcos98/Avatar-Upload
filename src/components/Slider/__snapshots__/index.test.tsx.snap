// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider component should be renders correctly 1`] = `
.c0 {
  position: relative;
}

.c3 {
  width: 100%;
  position: relative;
  -webkit-appearance: none;
  height: 2px;
  background-color: transparent;
  z-index: 2;
}

.c3::-webkit-slider-thumb {
  -webkit-appearance: none;
  height: 13px;
  width: 13px;
  background-color: #3F80FF;
  border-radius: 50%;
}

.c3:focus {
  outline-color: #3F80FF;
  outline-offset: 2px;
}

.c1 {
  position: absolute;
  z-index: 0;
  top: 13px;
  width: 100%;
  height: 2px;
  background-color: #B9D1FF;
  border-radius: 2px;
}

.c2 {
  display: block;
  width: 20%;
  height: 100%;
  background-color: #3F80FF;
  border-radius: 2px;
}

<div
  class="c0"
>
  <span
    class="c1"
  >
    <span
      class="c2"
    />
  </span>
  <input
    class="c3"
    max="60"
    min="10"
    type="range"
    value="20"
  />
</div>
`;
